### PR TITLE
feat(api): use new signature of RxJS' throwError

### PR DESCRIPTION
**Pull request by pchorus**
I changed `throwError` to use the new signature instead of the deprecated one.
Furthermore, I replaced `throwError` with a plain JS `throw new Error` since this is sufficient within RxJS operators.

### CODE REVIEW

**Review by dniggemann — APPROVED** [review]

File: modules/openapi-generator/src/main/resources/typescript-angular/api.service.mustache [hunk 1]
```diff
@@ -4,8 +4,10 @@
 import { Injectable, Optional } from '@angular/core';
 import { HttpClient, HttpHeaders, HttpParams, HttpParameterCodec } from '@angular/common/http';
 import { CustomHttpParameterCodec } from '../encoder';
-import { of, throwError } from 'rxjs';
-import { mergeMap } from 'rxjs/operators';
+// prettier-ignore
```
Comment (windwinkel) [inline modules/openapi-generator/src/main/resources/typescript-angular/api.service.mustache:7]: Why is this necessary now?

**Review by windwinkel — COMMENTED** [review]
One small thing, the rest looks good

File: modules/openapi-generator/src/main/resources/typescript-angular/api.service.mustache (hunk 1 above)
Comment (pchorus) [reply modules/openapi-generator/src/main/resources/typescript-angular/api.service.mustache:7]: Because the `throwError` is not always used in every api service. It is only used for operations (service functions) that contain parameters. That is not easily to check for the imports so I decided to simply ignore it.

**Review by windwinkel — APPROVED** [review]
